### PR TITLE
re-structure and clean up dockerfile

### PR DESCRIPTION
**Pull request by christian-heusel**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This PR increases the robustness of the production dockerfile as requested by @fosterfarrell9 😄 

* **What is the current behavior?** (You can also link to an open issue here)
The sidekiq assets are copied from a fixed path determined by the version

* **What is the new behavior (if this is a feature change)?**
The sidekiq assets are copied from the currently installed version

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
hopefully not!

### CODE REVIEW

File: docker/production/Dockerfile [hunk 1]
```diff
@@ -15,33 +15,34 @@ EXPOSE 3000
...
-
-
-RUN curl -sL https://deb.nodesource.com/setup_14.x | bash \
- && apt-get update && apt-get install -y nodejs && rm -rf /var/lib/apt/lists/* \
- && curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
- && echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list \
- && apt-get update && apt-get install -y yarn && rm -rf /var/lib/apt/lists/*
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        ffmpeg imagemagick pdftk ghostscript sqlite3 rsync shared-mime-info && \
```
Comment (tynsh) [inline docker/production/Dockerfile:20]: Why did we install sqlite? Can it be removed?

File: docker/production/Dockerfile [hunk 2]
```diff
@@ -15,33 +15,34 @@ EXPOSE 3000
...
+    rm -rf /var/lib/apt/lists/* && \
```
Comment (tynsh) [inline docker/production/Dockerfile:21]: This can be removed

File: docker/production/Dockerfile [hunk 3]
```diff
@@ -15,33 +15,34 @@ EXPOSE 3000
...
+    curl -sL https://deb.nodesource.com/setup_14.x | bash && \
+    apt-get update && \
+    apt-get install -y nodejs && \
+    rm -rf /var/lib/apt/lists/* && \
```
Comment (tynsh) [inline docker/production/Dockerfile:25]: This is also redundant

File: docker/production/Dockerfile [hunk 4]
```diff
@@ -15,33 +15,34 @@ EXPOSE 3000
...
 WORKDIR /usr/src/app
 USER app
 COPY --from=build-pdfcomprezzor /go/src/pdfcomprezzor.wasm  /usr/src/app/public/pdfcomprezzor/pdfcomprezzor.wasm
 
 COPY --chown=app:app ./Gemfile ./Gemfile.lock ./yarn.lock ./package.json /usr/src/app/
 
-RUN bundle install
-RUN yarn  install --production=false
+RUN bundle install && \
+    yarn  install --production=false
```
Comment (tynsh) [inline docker/production/Dockerfile:45]: Why did we disable production again?

File: docker/production/Dockerfile [hunk 5]
```diff
@@ -15,33 +15,34 @@ EXPOSE 3000
...
 COPY --chown=app:app . /usr/src/app
-RUN cp -r /usr/local/bundle/gems/sidekiq-6.2.0/web/assets /usr/src/app/public/sidekiq
-RUN SECRET_KEY_BASE="$(bundle exec rails secret)" DB_ADAPTER=nulldb   bundle exec rails assets:precompile
+RUN cp -r $(bundle info --path sidekiq)/web/assets /usr/src/app/public/sidekiq && \
```
Comment (tynsh) [inline docker/production/Dockerfile:47]: Looks sane

**Review by tynsh — APPROVED** [review]
So my opinion is, this is safe to merge and nothing changes. I just don't remember why we disabled yarn production mode and installed sqlite

File: docker/production/Dockerfile (hunk 1 above)
Comment (fosterfarrell9) [reply docker/production/Dockerfile:20]: It is probably a remnant from the first dockerfiles where we used sqlite in development.

Comment (fosterfarrell9) [reply docker/production/Dockerfile:20]: So it can be removed.

File: docker/production/Dockerfile (hunk 4 above)
Comment (fosterfarrell9) [reply docker/production/Dockerfile:45]: That is probably a question for @henrixapp .

Comment (henrixapp) [reply docker/production/Dockerfile:45]: I think webpacker and other dev-Deps would not not be installed, if this is `production=true`. We could move the devDeps in the package.json to dependencies and everything should work as expected.

File: docker/production/Dockerfile (hunk 2 above)
Comment (christian-heusel) [reply docker/production/Dockerfile:21]: Why can everything here be removed? 🤔

File: docker/production/Dockerfile (hunk 3 above)
Comment (christian-heusel) [reply docker/production/Dockerfile:25]: How is it redundant? 
According to [the script](https://deb.nodesource.com/setup_14.x) itself it just installs the repo and not the program:
```
#!/bin/bash

# Discussion, issues and change requests at:
#   https://github.com/nodesource/distributions
#
# Script to install the NodeSource Node.js 14.x repo onto a
# Debian or Ubuntu system.
#
# Run as root or insert `sudo -E` before `bash`:
#
# curl -sL https://deb.nodesource.com/setup_14.x | bash -
#   or
# wget -qO- https://deb.nodesource.com/setup_14.x | bash -
#
# CONTRIBUTIONS TO THIS SCRIPT
#
# This script is built from a template in
# https://github.com/nodesource/distributions/tree/master/deb/src
# please don't submit pull requests against the built scripts.
```

**Review by henrixapp — APPROVED** [review]

**Comment by henrixapp** [comment]
The one failing test is not caused by your code segment. Merge it at any time

File: docker/production/Dockerfile (hunk 3 above)
Comment (christian-heusel) [reply docker/production/Dockerfile]: Resolving because of missing feedback!

File: docker/production/Dockerfile (hunk 2 above)
Comment (christian-heusel) [reply docker/production/Dockerfile]: Resolving because of missing feedback!

**Pull request merged**